### PR TITLE
Alerting: Show Insights only for cloud users

### PR DESCRIPTION
**Pull request by VikaCep**
**What is this feature?**

- Creates common functions to detect whether the build version is OSS, and also if it's a development version. 
- Uses these functions to prevent showing the Insights landing page for OSS users 
- Uses these functions for the Declare Incident button 

**Why do we need this feature?**

Insights landing page uses datasources and metrics that are only available for cloud users, hence we need to restrict its access for that scenario only.

**Who is this feature for?**

All users

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/alerting-squad/issues/617

### CODE REVIEW

File: public/app/features/alerting/unified/home/Home.tsx [hunk 1]
```diff
@@ -56,7 +58,7 @@ export function getHomeApp(insightsEnabled: boolean) {
 }
 
 export default function Home() {
-  const insightsEnabled = !!config.featureToggles.alertingInsights;
+  const insightsEnabled = (!isOpenSourceEdition() || isDevelopment()) && !!config.featureToggles.alertingInsights;
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/home/Home.tsx:61]: Do we want to allow OSS users in development mode to be able to use this feature toggle?. If so, I would update the PR title and description. wdyt?

Comment (VikaCep) [reply public/app/features/alerting/unified/home/Home.tsx:61]: I believe the mode `development` is only used for local environments so it seemed safe to use here. I copied this from the Declare Incident button implementation. @gillesdemey could you confirm if the scenario that Sonia mentioned is possible?

Comment (gillesdemey) [inline public/app/features/alerting/unified/home/Home.tsx:61]: nit: prefer `Boolean()` over `!!<expression>`

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM!🚀

**Pull request merged**